pkg/state/impl/inmem/collection.go

```diff
@@ -455,10 +465,15 @@ func (collection *ResourceCollection) WatchAll(ctx context.Context, ch chan<- st
 			}
 
 			if collection.writePos-pos >= int64(collection.capacity) {
-				// buffer overrun, there's no way to signal error in this case,
-				// so for now just return
 				collection.mu.Unlock()
 
+				select {
+				case ch <- state.Event{
+					Type: state.Errored,
```

Comment:
why don't we fill the error field when the state is errored?

Reply:
:facepalm:  thanks for noticing that, I will write a test for it!

Reply:
I think the assertions need to be inverted as well :)